common/perf-test-core/src/main/java/com/azure/perf/test/core/TestProxyPolicy.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.perf.test.core;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelinePosition;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URI;
+
+public class TestProxyPolicy implements HttpPipelinePolicy {
+    private final URI uri;
+    
+    private String recordingId;
+    private String mode;
+
+    public TestProxyPolicy(URI uri) {
+        this.uri = uri;
+    }
+
+    public String getMode() {
+        return mode;
+    }
+
+    public void setMode(String mode) {
+        this.mode = mode;
+    }
+
+    public String getRecordingId() {
+        return recordingId;
+    }
+
+    public void setRecordingId(String recordingId) {
+        this.recordingId = recordingId;
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        if (recordingId != null && !recordingId.isEmpty() && mode != null && !mode.isEmpty()) {
```

Comment:
This can be refactored to `!CoreUtils.isNullOrEmpty(recordingId) && !CoreUtils.isNullOrEmpty(mode)